api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -48,6 +54,53 @@ message RedisProxy {
     bool enable_hashtagging = 2;
   }
 
-  // Network settings for the connection pool to the upstream cluster.
+  // Network settings for the connection pool to the upstream clusters.
   ConnPoolSettings settings = 3 [(validate.rules).message.required = true];
+
+  message PrefixRoutes {
+    message Route {
+      // String prefix that must match the beginning of the keys. Envoy will always favor the
+      // longest matches.
+      string prefix = 1 [(validate.rules).string.min_bytes = 1];
+
+      // Indicates if the prefix needs to be removed from the key when forwarded.
+      bool remove_prefix = 2;
+
+      // Upstream cluster to forward the command to.
+      string cluster = 3;
+    }
+
+    // List of prefix routes.
+    repeated Route routes = 1 [(gogoproto.nullable) = false];
+
+    // Indicates that prefix matching should be case insensitive.
+    bool case_insensitive = 2;
+
+    // Optional catch-all route to forward commands that doesn't match any of the routes. The
+    // catch-all route becomes required when no routes are specified.
+    string catch_all_cluster = 3;
+  }
+
+  // List of prefixes used to separate keys from different workloads to different clusters. Any
+  // overlapping prefixes will be ignored, and Envoy will always favor the longest matches first. A
```

Comment:
Instead of being ignored, can we just make it an error? Is there any reason not to do that? Also, is ignored the right phrase since in the example below, "abc" and "a" overlap?